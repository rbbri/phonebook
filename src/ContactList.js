import React, { Component } from 'react';

class ContactList extends Component {
  render () {
    return (
      <div className='ContactListMain'>
        Contacts
      </div>
    );
  }
}

export default ContactList;
