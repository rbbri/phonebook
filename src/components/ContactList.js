import React, { Component } from 'react';
import Contact from './Contact'

const API = 'http://www.mocky.io/v2/581335f71000004204abaf83'

class ContactList extends Component {
  state = {
    contacts: [],
    search: ""
  }

  componentDidMount(){
  fetch(API)
      .then(response => response.json())
      .then(data => this.setState({contacts: data.contacts}));
}

  changeSearch = e => {
    this.setState({search: e.target.value})
  }



  render() {

    const {search} = this.state;
    const contacts = this.state.contacts;
    const filteredContacts = contacts.filter(c => {
      return c.name.toLowerCase().indexOf( search.toLowerCase() ) !== -1
    })

    return (
      <div className='ContactListMain'>
        Contacts
        <div className='ContactsSearch'>
          <input type='text'
                 placeholder='Search'
                 onChange={this.changeSearch}
          />
        </div>
        <div className='Contacts'>
          {filteredContacts.map(c =>
            <Contact
              name={c.name}
              phoneNumber={c.phone_number}
              address={c.address}
            />
            )}
        </div>
      </div>
    );
  }
}

export default ContactList;
