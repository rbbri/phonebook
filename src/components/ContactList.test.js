import React, { Component } from 'react';
import ContactList from './ContactList.js'
import {shallow} from 'enzyme'
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });



test('renders without crashing', () => {
  shallow(<ContactList />);
});

test('renders correctly', () => {
  let wrapper = shallow(<ContactList />);
  expect(wrapper).toMatchSnapshot();
});
