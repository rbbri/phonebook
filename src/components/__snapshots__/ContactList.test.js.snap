// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContactList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
           Contacts 
        </h1>,
        <div
          className="ContactsSearch"
        >
          <input
            onChange={[Function]}
            placeholder="Search"
            type="text"
          />
        </div>,
        <div
          className="ContactsSort"
        >
          <select
            onChange={[Function]}
            value="Sort"
          >
            <option
              value="Sort"
            >
              Sort
            </option>
            <option
              value="Ascending"
            >
              Ascending
            </option>
            <option
              value="Descending"
            >
              Descending
            </option>
          </select>
        </div>,
        <div
          className="Contacts"
        />,
      ],
      "className": "ContactListMain",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Contacts ",
        },
        "ref": null,
        "rendered": " Contacts ",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            onChange={[Function]}
            placeholder="Search"
            type="text"
          />,
          "className": "ContactsSearch",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Search",
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <select
            onChange={[Function]}
            value="Sort"
          >
            <option
              value="Sort"
            >
              Sort
            </option>
            <option
              value="Ascending"
            >
              Ascending
            </option>
            <option
              value="Descending"
            >
              Descending
            </option>
          </select>,
          "className": "ContactsSort",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <option
                value="Sort"
              >
                Sort
              </option>,
              <option
                value="Ascending"
              >
                Ascending
              </option>,
              <option
                value="Descending"
              >
                Descending
              </option>,
            ],
            "onChange": [Function],
            "value": "Sort",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Sort",
                "value": "Sort",
              },
              "ref": null,
              "rendered": "Sort",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Ascending",
                "value": "Ascending",
              },
              "ref": null,
              "rendered": "Ascending",
              "type": "option",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Descending",
                "value": "Descending",
              },
              "ref": null,
              "rendered": "Descending",
              "type": "option",
            },
          ],
          "type": "select",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "Contacts",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
             Contacts 
          </h1>,
          <div
            className="ContactsSearch"
          >
            <input
              onChange={[Function]}
              placeholder="Search"
              type="text"
            />
          </div>,
          <div
            className="ContactsSort"
          >
            <select
              onChange={[Function]}
              value="Sort"
            >
              <option
                value="Sort"
              >
                Sort
              </option>
              <option
                value="Ascending"
              >
                Ascending
              </option>
              <option
                value="Descending"
              >
                Descending
              </option>
            </select>
          </div>,
          <div
            className="Contacts"
          />,
        ],
        "className": "ContactListMain",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Contacts ",
          },
          "ref": null,
          "rendered": " Contacts ",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              onChange={[Function]}
              placeholder="Search"
              type="text"
            />,
            "className": "ContactsSearch",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Search",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <select
              onChange={[Function]}
              value="Sort"
            >
              <option
                value="Sort"
              >
                Sort
              </option>
              <option
                value="Ascending"
              >
                Ascending
              </option>
              <option
                value="Descending"
              >
                Descending
              </option>
            </select>,
            "className": "ContactsSort",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <option
                  value="Sort"
                >
                  Sort
                </option>,
                <option
                  value="Ascending"
                >
                  Ascending
                </option>,
                <option
                  value="Descending"
                >
                  Descending
                </option>,
              ],
              "onChange": [Function],
              "value": "Sort",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Sort",
                  "value": "Sort",
                },
                "ref": null,
                "rendered": "Sort",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Ascending",
                  "value": "Ascending",
                },
                "ref": null,
                "rendered": "Ascending",
                "type": "option",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Descending",
                  "value": "Descending",
                },
                "ref": null,
                "rendered": "Descending",
                "type": "option",
              },
            ],
            "type": "select",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "Contacts",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
